brave/browser-laptop
# Allow unblocking of scripts from urls containing ip addresses

Auditors: @diracdeltas

Fixes #2988

Test Plan:
1. Create a temporary folder with `index.html` containing:
```
<html>
<p> welcome 2 mi web sight </p>
<p id="sneaky" style="display:none"> y u allow js </p>
<script>
document.getElementById('sneaky').style.display='block';
</script>
</html>
````
2. Run `python -m SimpleHTTPServer` inside the temporary folder.
3. Open Brave, go to Preferences and disable scripts globally.
3. Open the url `http://127.0.0.1:8000`.
4. Verify unblocking the script adds the additional text `y u allow js`.

I also added automated tests to the noscript info tests which can be run via:
```
npm run test -- --grep="^noscript"
```
